src/features/content-explorer/content-explorer/__tests__/ContentExplorer.test.js

```diff
@@ -453,6 +453,75 @@ describe('features/content-explorer/content-explorer/ContentExplorer', () => {
 
             expect(onSelectItemSpy.notCalled).toBe(true);
         });
+
+        test('should set isSelectAllChecked to true if all displayed items are selected and isSelectAllAllowed is true', () => {
+            const clickedItemIndex = 0;
+            const clickedItemIndex2 = 1;
+            const clickedItemIndex3 = 2;
+            const items = [
+                { id: '1', name: 'item1' },
+                { id: '2', name: 'item2' },
+                { id: '3', name: 'item3' },
+            ];
+            const isSelectAllAllowed = true;
+            const wrapper = renderComponent(
+                {
+                    items,
+                    isSelectAllAllowed,
+                    onSelectItem: onSelectItemSpy,
+                    contentExplorerMode: ContentExplorerModes.MULTI_SELECT,
+                },
+                true,
+            );
+
+            wrapper
+                .find('.table-row')
+                .at(clickedItemIndex)
+                .simulate('click');
+
+            wrapper
+                .find('.table-row')
+                .at(clickedItemIndex2)
+                .simulate('click');
+
+            wrapper
+                .find('.table-row')
+                .at(clickedItemIndex3)
+                .simulate('click');
+
+            expect(Object.keys(wrapper.state('selectedItems')).length).toEqual(3);
+            expect(wrapper.state('isSelectAllChecked')).toBe(true);
+        });
+
+        test('should set isSelectAllChecked to false if an item is unchecked while isSelectAllChecked is true', () => {
+            const clickedItemIndex = 0;
+            const clickedItemIndex2 = 1;
+            const items = [
+                { id: '1', name: 'item1' },
+                { id: '2', name: 'item2' },
+            ];
+            const selectedItems = { '1': items[clickedItemIndex], '2': items[clickedItemIndex2] };
+            const isSelectAllAllowed = true;
+            const isSelectAllChecked = true;
```

Comment:
nit thing: this might be more preference but I think single-use variables intended for objects don't provide additional value over using the literal value directly in the object.

something like
```
const nextValue = 123;
[...]
const object = { nextValue };
```

can sometimes be worse than
```
const object = { nextValue: 123 };
```

because now the developer needs to find where the variable was set

---

so when I was reading
```
wrapper.setState({ selectedItems, isSelectAllChecked });
```
my first impression was `isSelectAllChecked` was dynamic and I had to scroll up to see that the value was  fixed to `true`

Reply:
That is a fair point. I will change to using the literal value in the object.